### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by 4iner**
Authorization & Authentication working, comment if you have any concerns

### CODE REVIEW

**Comment by 4iner** [comment]
Will need a Redis server running in the background to test the session (runs localhost:6379 by default)
can install here: 
https://github.com/microsoftarchive/redis/releases/download/win-3.2.100/Redis-x64-3.2.100.msi

**Comment by dillonclaremont** [comment]
The Travis CI builds have failed, is this related to the authentication issue or the code changes in this PR?

File: src/main/java/security/UserServiceDetails.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    
+    @Autowired
+    CustomerRepository customerRepository;
+
+    @Autowired
+    BookstoreOwnerRepository bookstoreOwnerRepository;
+
+    @Autowired
+    public UserServiceDetails(CustomerRepository customerRepository, BookstoreOwnerRepository bookstoreOwnerRepository) {
+        // this instantiates a couple users to start out with
```
Comment (dillonclaremont) [inline src/main/java/security/UserServiceDetails.java]: Why is this necessary?

File: src/main/java/security/UserServiceDetails.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.stereotype.Service;
+
+import models.BookstoreOwner;
+import models.BookstoreUser;
+import models.Customer;
+import repositories.BookstoreOwnerRepository;
+import repositories.CustomerRepository;
+
+@Service
+public class UserServiceDetails implements UserDetailsService {
```
Comment (dillonclaremont) [inline src/main/java/security/UserServiceDetails.java]: What is the purpose of this class?

File: .travis.yml [hunk 3]
```diff
@@ -3,4 +3,6 @@ deploy:
   provider: heroku
   api-key:
     secure: $HEROKU_API_KEY
-  app: sysc-4806-project-2020
\ No newline at end of file
+  app: sysc-4806-project-2020
+services:
+  - redis-server
```
[the remark this answers is not in the document]
Comment (4iner) [reply .travis.yml]: Stores session data in-memory for users

**Comment by dillonclaremont** [comment]
Let’s hold on this PR until we have the new UI deployable on heroku or have adapted the dev portal to have the ability to authenticate users

File: src/main/java/controllers/BookstoreRestController.java [hunk 4]
```diff
@@ -55,8 +74,8 @@ public BookstoreOwner getBookstoreOwner(@RequestParam(value = "bookstoreOwnerId"
      * @return Newly created BookstoreOwner
      */
     @PostMapping("/api/newBookstoreOwner")
-    public BookstoreOwner newBookstoreOwner(@RequestParam(value = "bookstoreOwnerName") String bookstoreOwnerName) {
-        BookstoreOwner bookstoreOwner = new BookstoreOwner(bookstoreOwnerName);
+    public BookstoreOwner newBookstoreOwner(@RequestParam(value = "bookstoreOwnerName") String bookstoreOwnerName, @RequestParam(value = "bookstoreOwnerUsername") String bookstoreOwnerUsername, @RequestParam(value = "bookstoreOwnerPassword") String bookstoreOwnerPassword) {
```
Comment (thomasbryk) [inline src/main/java/controllers/BookstoreRestController.java]: Update java doc for this method to include new params

File: src/main/java/controllers/BookstoreRestController.java [hunk 5]
```diff
@@ -228,8 +249,8 @@ public Customer getCustomer(@RequestParam(value = "customerId") long customerId)
      * @return Newly created Customer
      */
     @PostMapping("/api/newCustomer")
-    public Customer newCustomer(@RequestParam(value = "customerName") String customerName, @RequestParam(value = "address") String address, @RequestParam(value = "email") String email, @RequestParam(value = "phoneNumber") String phoneNumber) {
-        Customer customer = new Customer(customerName, address, email, phoneNumber);
+    public Customer newCustomer(@RequestParam(value = "customerName") String customerName, @RequestParam(value = "address") String address, @RequestParam(value = "email") String email, @RequestParam(value = "phoneNumber") String phoneNumber, @RequestParam(value = "username") String username, @RequestParam(value = "password") String password) {
```
Comment (thomasbryk) [inline src/main/java/controllers/BookstoreRestController.java]: Update java doc for this method to include new params

**Comment by dillonclaremont** [comment]
Working on updating unit tests, since those were merged into milestone-2, will create a branch off of this feature branch with fixes to tests to accommodate your changes and create a PR into this branch

**Comment by dillonclaremont** [comment]
Let's get a Heroku build before we merge to milestone2, to confirm Redis functionality on Heroku!

**Comment by dillonclaremont** [comment]
Heroku deploy test was successful!
